certificatelock.php

```diff
@@ -43,8 +43,19 @@
     if ($form->is_submitted()) {
 
         $certfiles = array($saml2auth->certpem, $saml2auth->certcrt);
-        foreach ($certfiles as $certfile) {
-            chmod($certfile, 0440);
+        if (isset($data->unlockcertsbutton)) {
+            // Change the permissions in order to regenerate if unlocked.
+            foreach ($certfiles as $certfile) {
+                chmod($certfile, 0660);
```

Comment:
there is a moodle setting which I think we should use for this

$CFG->filepermissions